### DISCUSSION TITLE
Changing AD material property variable to regular variable

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#24215] [Q&A General] [answered]
Hello,
I’m retrieving an ADmaterial property, but I want to strip the AD from it and convert it to a regular property before using it in my code. How do I do this? I tried using the `.value()` function but got this error: 
```
error: 'const class ADMaterialProperty<RankTwoTensorTempl<double> >' has no member named 'value'; did you mean '_value'?
 132 |         RankTwoTensor _stress = (_stress_o).value();
     |                                             ^~~~~
     |                                             _value
/home/jay/MOOSE/projects/creep/src/materials/perm.C:133:16: error: no match for 'operator[]' (operand types are 'RankTwoTensor' {aka 'RankTwoTensorTempl<double>'} and 'unsigned int')
 133 |         _stress[_qp].symmetricEigenvaluesEigenvectors(eigvals, eigvec);
     |                ^
In file included from /home/jay/MOOSE/projects/creep/build/unity_src/materials_Unity.C:4:
/home/jay/MOOSE/projects/creep/src/materials/perm2.C: In member function 'void perm2Templ<is_ad>::computeQpProperties()':
/home/jay/MOOSE/projects/creep/src/materials/perm2.C:93:9: error: '_stress' was not declared in this scope; did you mean '_stress_o'?
  93 |         _stress[_qp].symmetricEigenvaluesEigenvectors(eigvals, eigvec);
```

I have the following in my header and source:
`const ADMaterialProperty<RankTwoTensor>& _stress;`

`_stress_o(getADMaterialProperty<RankTwoTensor>(_base_name + "stress")),`

My code:
`RankTwoTensor _stress[_qp] = (_stress_o[_qp]).value();`

Do I need to include any special type of header? Thanks.

**Comment by GiudGiud** [comment]
this should be fine
(_stress_o[_qp]).value()

but 
(_stress_o).value(); is not, you need to index to get the ADReal before stripping the derivative.

Note that there is already a converter between AD and non AD:
https://mooseframework.inl.gov/source/materials/MaterialADConverter.html

**Reply by hugary1995** [reply, marked as answer]
or `raw_value(_stress_o[_qp])`

**Reply by steam-simulator** [reply]
Thank you very much, @hugary1995 and @GiudGiud , for the recommendations. In this case, `raw_value(_stress_o[_qp])` works best!
